### DISCUSSION TITLE
Does reboot makes the carrier attribute a new public IP address to the modem?

### DISCUSSION THREAD

**Discussion by sergioleite** [Salamek/huawei-lte-api#180] [Q&A] [answered]
I'm writing a web-scrapping application in Python which has to bypass a very strict throtelling website.

Currently I'm testing the solution with an Android phone as the gateway to the Internet, so every time I toggle the airplane-mode, off and on, the carrier attributes a new IP to the phone, so I can make a new http request to the website without being blocked.

After getting to know huawei-lte-api, I'd like to know, before buying a compatible stick modem, if when I call the reboot method a new IP address is attributed by the carrier to the modem?

Thanks in advance folks!

**Comment by Salamek** [comment, marked as answer]
IP assignment is fully under connection provider control so IP assignment rules should be same for all devices connected to their network.
(We assume that you are unable to spoof your IMEI and that you did not request static IP from provider) 

Also you won't need to reboot the device, only thing you need is to toggle LTE interface to reconnect

**Reply by sergioleite** [reply]
Thank you for the clarification @Salamek !
